module/VuFind/src/VuFind/Db/Service/ChangeTrackerService.php

```diff
@@ -54,9 +54,9 @@ class ChangeTrackerService extends AbstractService implements LoggerAwareInterfa
      * @param string $core The Solr core holding the record.
      * @param string $id   The ID of the record being indexed.
      *
-     * @return ?ChangeTracker
+     * @return ChangeTracker|false
      */
-    public function retrieve(string $core, string $id): ?ChangeTracker
+    public function retrieve(string $core, string $id): ChangeTracker|false
```

Comment:
I wonder if, rather than changing the return type, it would be more elegant to convert a falsy return value from `current` into null... e.g.:

```
$queryResult = $query->getResult();
$result = current($queryResult);
return $result ? $result : null;
```

If you prefer to leave it as-is, I don't particularly object, but I think using nulls here might be a little cleaner.

It might also be more clear if we use `ChangeTracker|false` for methods that return false when they encounter an error condition, and `?ChangeTracker` for methods that return an object when a record exists and null when it does not. This way, `null` can mean "no data" and `false` can mean "something went wrong."